cheroot/server.py

```diff
@@ -598,12 +599,17 @@ class HTTPRequest(object):
     A HeaderReader instance or compatible reader.
     """
 
-    def __init__(self, server, conn):
+    def __init__(self, server, conn, proxy_mode=False, strict_mode=True):
```

Comment:
I see your point, but currently I cannot suggest cleaner solution. So for now I'm going to merge this as is.